tests/test_lessonbuilder_fix_special_chars.py

```diff
@@ -0,0 +1,40 @@
+import os
+import shutil
+import unittest
+import argparse
+import config
+import config.config
+from work.lessonbuilder_rewrite_urls import fix_unwanted_url_chars
+from unittest.mock import patch
+import xml.etree.ElementTree as ET
+from bs4 import BeautifulSoup
+
+class LessonbuilderUpdateUrlRewriteTestCase(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+
+    @patch('os.rename')
+    @patch('argparse.ArgumentParser.parse_args', return_value=argparse.Namespace(SITE_ID='123456', debug=True))
+
+    def test_url_rewrite_chars_fix(self, *_):
+        xml_src = self.ROOT_DIR + '/test_files/site_123456-archive/lessonbuilder.xml'
+        xml_old = self.ROOT_DIR + '/test_files/site_123456-archive/lessonbuilder.old'
+        self.check_urls_chars(xml_src=xml_src)
+        shutil.move(xml_old, xml_src)
+    def check_urls_chars(self, xml_src: str):
+
+        tree = ET.parse(xml_src)
+        root = tree.getroot()
+
+        urlprefix = 'https://vula.uct.ac.za/access/content/group/site_123456'
+        expected_outcome = '../Lesson 1 with in title/Jpeg_thumb_artifacts_test.jpg'
+
+        for item in root.findall(".//item[@type='5']"):
+            html = BeautifulSoup(item.attrib['html'], 'html.parser')
+            for element in html.find_all(attrs={'src': True}):
+                current_url = element.get('src')
+                self.assertEquals((fix_unwanted_url_chars(current_url, urlprefix)), expected_outcome)
```

Comment:
This is kind of a mixture of two techniques. You're doing all the setup by providing a lessonbuilder.xml but then not actually calling the full workflow that would run on it. 

Modify what you have here to call the full operation on lessonbuilder.xml, then you parse the updated lessonbuilder.xml to get the resulting src element and compare that to expected_outcome (without calling fix_unwanted_url_chars explicitly because that should have been called by the full operation).

Then to test a wider range of options without the overhead of putting them all into a sample lessonbuilder.xml, you call separately

                self.assertEquals((fix_unwanted_url_chars(input_url, urlprefix)), expected_outcome)

with different values for input_url and expected_outcome.